test/check/check.go

```diff
@@ -414,3 +416,49 @@ func configToYaml(config *opv1alpha1.NetworkAddonsConfig) string {
 	}
 	return string(manifest)
 }
+
+// CheckUnicast return an error in case that the given addresses support multicast or is invalid.
+func CheckUnicastAndValidity() (string, string) {
+	rangeStart, rangeEnd := retrieveRange()
+	parsedRangeStart, err := net.ParseMAC(rangeStart)
+	Expect(err).ToNot(HaveOccurred())
+	checkUnicast(parsedRangeStart)
+
+	parsedRangeEnd, err := net.ParseMAC(rangeEnd)
+	Expect(err).ToNot(HaveOccurred())
+	checkUnicast(parsedRangeEnd)
+	return rangeStart, rangeEnd
+}
+
+func checkUnicast(mac net.HardwareAddr) {
+	// A bitwise AND between 00000001 and the mac address first octet.
+	// In case where the LSB of the first octet (the multicast bit) is on, it will return 1, and 0 otherwise.
+	multicastBit := 1 & mac[0]
+	Expect(multicastBit).ToNot(BeNumerically("==", 1), "invalid mac address. Multicast addressing is not supported. Unicast addressing must be used. The first octet is %#0X", mac[0])
+}
+
+func retrieveRange() (string, string) {
+	configMap := &corev1.ConfigMap{}
+	// KubeMacPool is known to restart shortly after first started, try to skip this initial restart
```

Comment:
I don't think this is related to the bug. Yes, the pod is being restarted, but the config map just sits there after KubeMacPool is installed. The reason why you have to keep Eventually I think is, that you don't wait until the components is installed in BeforeEach